### ISSUE TITLE
Jedis 2.9.3 leaking connections causing the Pool to exhaust

### ISSUE THREAD

**Issue by NitishBangera** [xetorthio/jedis#2047]
### Expected behavior

No connection leak

### Actual behavior

There's a connection leak with heavy usage. The application runs fine for 3-4 hours and then it all the connections wait infinitely(Thread dump) and the application throws a Timeout on wait exception.

### Steps to reproduce:

Tested with Jmeter calling a rest API which internally does a get to Redis using Jedis 2.9.3. 200 concurrent users doing 10 calls per user to the rest API.

### Redis / Jedis Configuration
Jedis Pool:
max_total: 1500
max_idle: 500
min_idle: 10
max_wait: 100 ms

#### Jedis version:
2.9.3

#### Redis version:
5.0.3

#### Java version:
Jdk8 version 91

**Comment by NitishBangera** [comment]
```
"ForkJoinPool.commonPool-worker-1" #3305 daemon prio=5 os_prio=0 tid=0x00007fa458117800 nid=0x3f1357 runnable [0x00007fa5e8ac9000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x00000000c059fe78> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
        at org.apache.commons.pool2.impl.LinkedBlockingDeque.pollFirst(LinkedBlockingDeque.java:638)
        at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:427)
        at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:346)
        at redis.clients.util.Pool.getResource(Pool.java:49)
        at redis.clients.jedis.JedisPool.getResource(JedisPool.java:226)
        at redis.clients.jedis.JedisPool.getResource(JedisPool.java:16)
        at org.springframework.data.redis.connection.jedis.JedisConnectionFactory.fetchJedisConnector(JedisConnectionFactory.java:271)
        at org.springframework.data.redis.connection.jedis.JedisConnectionFactory.getConnection(JedisConnectionFactory.java:464)
        at org.springframework.data.redis.core.RedisConnectionUtils.doGetConnection(RedisConnectionUtils.java:132)
        at org.springframework.data.redis.core.RedisConnectionUtils.bindConnection(RedisConnectionUtils.java:70)
        at org.springframework.data.redis.core.RedisTemplate.execute(RedisTemplate.java:209)
        at org.springframework.data.redis.core.RedisTemplate.execute(RedisTemplate.java:184)
```
Is this issue resolved in Jedis 2.10.2?

**Comment by NitishBangera** [comment]
Looks like Jedis 2.10.2 solves the issue.
